theseus/optimizer/nonlinear/nonlinear_least_squares.py

```diff
@@ -32,21 +97,404 @@ def __init__(
         step_size: float = 1.0,
         **kwargs,
     ):
-        linear_solver_cls = linear_solver_cls or CholeskyDenseSolver
         super().__init__(
             objective,
             linear_solver_cls=linear_solver_cls,
             vectorize=vectorize,
-            linearization_cls=linearization_cls,
-            linearization_kwargs=linearization_kwargs,
-            linear_solver_kwargs=linear_solver_kwargs,
             abs_err_tolerance=abs_err_tolerance,
             rel_err_tolerance=rel_err_tolerance,
             max_iterations=max_iterations,
             step_size=step_size,
             **kwargs,
         )
+        linear_solver_cls = linear_solver_cls or CholeskyDenseSolver
+        linear_solver_kwargs = linear_solver_kwargs or {}
+        self.linear_solver = linear_solver_cls(
+            objective,
+            linearization_cls=linearization_cls,
+            linearization_kwargs=linearization_kwargs,
+            **linear_solver_kwargs,
+        )
+        self.ordering = self.linear_solver.linearization.ordering
+        self._tmp_optim_vars = tuple(v.copy(new_name=v.name) for v in self.ordering)
+
+    # Modifies the (no grad) info in place to add data of grad loop info
+    def _merge_infos(
```

Comment:
Could this be a fn in the NonLinearOptimizer class instead? I need this fn in GBP so the alternative is that I copy this code into the GBP class.

Reply:
@dishank-b, @bamos does this function get similarly used for CEM? If so then we could move it one level up. If not then we can figure out the common intersection for this function between NLS, GBP and CEM.